timesketch/api/v1/resources/sigma.py

```diff
@@ -0,0 +1,220 @@
+# Copyright 2020 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Sigma resources for version 1 of the Timesketch API."""
+
+from flask import abort
+from flask_restful import Resource
+from flask_restful import reqparse
+from flask_login import login_required
+from flask_login import current_user
+from flask import jsonify
+from flask import current_app
+
+import six
+import os
+import codecs
+import yaml
+
+from timesketch.api.v1 import resources
+from timesketch.lib.definitions import HTTP_STATUS_CODE_OK
+from timesketch.lib.definitions import HTTP_STATUS_CODE_CREATED
+from timesketch.lib.definitions import HTTP_STATUS_CODE_BAD_REQUEST
+from timesketch.lib.definitions import HTTP_STATUS_CODE_FORBIDDEN
+from timesketch.lib.definitions import HTTP_STATUS_CODE_NOT_FOUND
+
+from sigma.backends import elasticsearch as sigma_elasticsearch
+import sigma.configuration as sigma_configuration
+from sigma.parser import collection as sigma_collection
+
+
+# TODO maybe do that moving forward and move code from here to the model
+#from timesketch.models.sigma import Sigma
+from timesketch.lib.analyzers import interface
```

Comment:
this should absolutely not be imported here....

if you need to ikmport something from the analyzer interface, it shouldn't be there, but in a different file that both this file and the analyzer interface imports....

Reply:
ACK